### PR TITLE
Fix logic that sets edit link visibility #11213

### LINKED ISSUE #11213: Edit link not visible in search results using default allow
The edit link is not availible in search results even if a user has write permissions on an instance.

### PR DESCRIPTION
**Pull request by chiatt**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change
Displays edit link in search results if user has edit permissions on an instance.

### Issues Solved

Closes #11213

### Checklist

-   I targeted one of these branches:
    - [x] dev/7.6.x (under development): features, bugfixes not covered below
    - [ ] dev/7.5.x (main support): regressions, crashing bugs, security issues, major bugs in new features
    - [ ] dev/6.2.x (extended support): major security issues, data loss issues
-   [ ] I added a changelog in arches/releases
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [ ] Unit tests pass locally with my changes
-   [ ] I added tests that prove my fix is effective or that my feature works
-   [ ] My test fails on the target branch

#### Accessibility Checklist

[Developer Guide](https://arches.readthedocs.io/en/stable/developing/advanced/accessibility/)

| Topic            | Changed | Retested |
| ---------------- | ------- | -------- |
| Color contrast   |         |          |
| Form fields      |         |          |
| Headings         |         |          |
| Links            |         |          |
| Keyboard         |         |          |
| Responsive Design|         |          |
| HTML validation  |         |          |
| Screen reader    |         |          |

#### Ticket Background
*   Sponsored by: 
*   Found by: @ 
*   Tested by: @ 
*   Designed by: @

### CODE REVIEW

**Review by aarongundel — APPROVED** [review]
Makes sense.

**Review by aarongundel — CHANGES_REQUESTED** [review]
Looks like there's a black formatting issue detected here.  Other than that, this makes sense.

**Comment by chrabyrd** [comment]
This is blocking some resource instance lifecycle work, so I'm adding myself as reviewer/assignee after a chat with Aaron 👍

File: arches/app/views/search.py [hunk 1]
```diff
@@ -393,6 +393,7 @@ def search_results(request, returnDsl=False):
     dsl.include("displaydescription")
     dsl.include("map_popup")
     dsl.include("provisional_resource")
+    dsl.include("permissions")
```
Comment (chrabyrd) [inline arches/app/views/search.py:396]: `permissions` was not available in the `get_search_ui_permissions` method

**Review by chrabyrd — APPROVED** [review]
lgtm 👍

File: arches/app/permissions/arches_standard.py [hunk 2]
```diff
@@ -598,13 +598,9 @@ def get_editable_resource_types(self, user: User) -> list[str]:
         user -- the user to check
 
         """
-
-        if self.user_is_resource_editor(user):
```
[the remark this answers is not in the document]
Comment (chrabyrd) [reply arches/app/permissions/arches_standard.py]: after a chat w/Cyrus I understand this is the intended behavior.

**Pull request merged**